src/DTIntrospection.cpp

```diff
@@ -0,0 +1,661 @@
+                featuresAndValues[i].insert(0, featureNames[i] + "=");
+            }
+            
+            for (const auto& item : featuresAndValues) {
+                FeatureOpValue featureOpValue2 = extractFeatureOpValue(item);
+                double value2AsDouble = convert(featureOpValue2.value);
+                
+                if (!std::isnan(valueAsDouble) && !std::isnan(value2AsDouble)) {
+                    if (featureOpValue.feature == featureOpValue2.feature && value2AsDouble <= valueAsDouble) {
+                        samples.push_back(sample);
+                        break;
+                    }
+                }
+
+            }
+            
+        }
+    }
+    else if (featureOpValue.op == ">") {
+        for (const auto& samplePair : trainingDataDict) {
+            const int sample = samplePair.first;
+            vector<string> featuresAndValues = samplePair.second;
+
+            for (int i = 0; i < featuresAndValues.size(); i++) {
+                featuresAndValues[i].insert(0, featureNames[i] + "=");
+            }
+            
+            for (const auto& item : featuresAndValues) {
+                FeatureOpValue featureOpValue2 = extractFeatureOpValue(item);
+                double value2AsDouble = convert(featureOpValue2.value);
+                
+                if (!std::isnan(valueAsDouble) && !std::isnan(value2AsDouble)) {
+                    if (featureOpValue.feature == featureOpValue2.feature && value2AsDouble > valueAsDouble) {
+                        samples.push_back(sample);
+                        break;
+                    }
+                }
+
+            }
+            
+        }
+    }
+    else {
+        throw std::runtime_error("Something is wrong with the feature-value syntax");
+    }
+    
+    return samples;
+}
+
+/**
+ * @brief Extracts the feature, operator, and value from a given feature value combination string.
+ *
+ * This function takes a string representing a feature value combination and extracts the feature name,
+ * the operator (either '=', '<', or '>'), and the value. The input string should be in one of the following formats:
+ * - "feature=value"
+ * - "feature<value"
+ * - "feature>value"
+ *
+ * If the input string does not match any of these formats, the function throws a runtime error.
+ *
+ * @param featureValueCombo The input string representing the feature value combination.
+ * @return FeatureOpValue A struct containing the extracted feature, operator, and value.
+ * @throws std::runtime_error If the input string does not match any of the expected formats.
+ */
+FeatureOpValue DTIntrospection::extractFeatureOpValue(string featureValueCombo) {
+    std::regex pattern1(R"((.+)=(.+))");
+    std::regex pattern2(R"((.+)<(.+))");
+    std::regex pattern3(R"((.+)>(.+))");
+
+    string feature, op, value;
+    std::smatch match;
+
+
+    if (std::regex_search(featureValueCombo, match, pattern1)) {
```

Comment:
We discussed this a bit, probably faster to not use regex but not a huge deal.

Reply:
Yes, could be an optimization in the future, but for now, it works.